test/abstract_client.js

```diff
@@ -450,56 +449,70 @@ module.exports = function (server, config) {
       })
     })
 
-    it('should delay ending up until all inflight messages are delivered', function (done) {
-      var client = connect()
-      var subscribeCalled = false
+    it('should call cb if store.put fails', function (done) {
+      const store = new Store()
+      store.put = function (packet, cb) {
+        process.nextTick(cb, new Error('oops there is an error'))
+      }
+      var client = connect({ incomingStore: store, outgoingStore: store })
+      client.publish('test', 'test', { qos: 2 }, function (err) {
+        if (err) {
+          client.end(true, done)
+        }
+      })
+    })
 
-      client.on('connect', function () {
-        client.subscribe('test', function () {
-          subscribeCalled = true
-        })
-        client.publish('test', 'test', function () {
-          client.end(false, function () {
-            subscribeCalled.should.be.equal(true)
-            done()
```

Comment:
why all these changes in the test? Can you only add the new one?

Reply:
I did not add any of those it only got indented. The test did not run otherwise

Reply:
Can you please restore them as they where on master?

Reply:
umm okay.. so you want me to revert the commit ?

Reply:
I don't get what I should be doing given that the current master test/abstract-client has many more changes in it. Should I simply revert the merge and commit it again with my addition in the test script?

Reply:
Yes, that would be the preferred path.

Reply:
okay so I reverted the merge and checked in my code again.